src/aof.c

```diff
@@ -1413,8 +1413,10 @@ int loadSingleAppendOnlyFile(char *filename) {
      * to the same file we're about to read. */
     server.aof_state = AOF_OFF;
 
-    client *old_client = server.current_client;
-    fakeClient = server.current_client = createAOFClient();
+    client *old_cur_client = server.current_client;
+    client *old_exec_client = server.executing_client;
+    fakeClient = createAOFClient();
+    server.current_client = server.executing_client = fakeClient;
```

Comment:
As memtioned in the top comment, `current_client` is external connection, but here it's set to fakeClient, which is a bit weird.

Reply:
you're right that it's odd..
it's the outer client that triggered the recursion.
maybe we can set it to NULL in that case (we don't do ACL and such).
or we can add a comment either here, or improve the one in server.h.
WDYT?

Reply:
I'm not sure if setting it to NULL is correct, because we can also trigger it with `debug loadaof`, when `current_client `should be the real connection.

Reply:
we can also leave it unset.. i think that if the startup loading has it set to NULL, it's safer to have DEBUG LOADAOF do the same (so the tests that use it, have the same state as they do in startup).

but still, i feel more comfortable add a comment and keeping the current code.

Reply:
@oranagra i think you forgot to add that comment